# Configuration parameters

# Robot wireframe size parameters
hip_link_length: 0.055
upper_leg_link_length: 0.1075
lower_leg_link_length: 0.130
body_width: 0.078
body_length: 0.186

# Stance parameters
default_stand_height: 0.155
stand_front_x_offset: 0.015
stand_back_x_offset: -0.000
lie_down_height: 0.083
lie_down_foot_x_offset: 0.065

# Servo parameters
num_servos: 12
servo_max_angle_deg: 270.0
RF_3: {num: 1,  center: 600, range: 400, direction:  1, center_angle_deg: 135.0, max_angle_deg: 270.0}
RF_2: {num: 2,  center: 600, range: 400, direction:  1, center_angle_deg: 135.0, max_angle_deg: 270.0}
RF_1: {num: 3,  center: 600, range: 400, direction: -1, center_angle_deg: 135.0, max_angle_deg: 270.0}
RB_3: {num: 4,  center: 600, range: 400, direction:  1, center_angle_deg: 135.0, max_angle_deg: 270.0}
RB_2: {num: 5,  center: 600, range: 400, direction:  1, center_angle_deg: 135.0, max_angle_deg: 270.0}
RB_1: {num: 6,  center: 600, range: 400, direction: -1, center_angle_deg: 135.0, max_angle_deg: 270.0}
LF_3: {num: 7,  center: 600, range: 400, direction:  1, center_angle_deg: 135.0, max_angle_deg: 270.0}
LF_2: {num: 8,  center: 600, range: 400, direction:  1, center_angle_deg: 135.0, max_angle_deg: 270.0}
LF_1: {num: 9,  center: 600, range: 400, direction: -1, center_angle_deg: 135.0, max_angle_deg: 270.0}
LB_3: {num: 10, center: 600, range: 400, direction:  1, center_angle_deg: 135.0, max_angle_deg: 270.0}
LB_2: {num: 11, center: 600, range: 400, direction:  1, center_angle_deg: 135.0, max_angle_deg: 270.0}
LB_1: {num: 12, center: 600, range: 400, direction: -1, center_angle_deg: 135.0, max_angle_deg: 270.0}

# Gait parameters
max_fwd_velocity: 0.2
max_side_velocity: 0.2
max_yaw_rate: 0.5
z_clearance: 0.05
alpha: 0.5
beta: 0.5
num_phases: 4
rb_contact_phases: [1, 0, 1, 1]
rf_contact_phases: [1, 1, 1, 0]
lf_contact_phases: [1, 0, 1, 1]
lb_contact_phases: [1, 1, 1, 0]
overlap_time: 0.18
swing_time: 0.18
body_shift_phases: [0, 0, 0, 0]
fwd_body_balance_shift: 0.0
back_body_balance_shift: 0.015
side_body_balance_shift: 0.015

# TF Parameter
lidar_x_pos: 0.045
lidar_y_pos: 0.0
lidar_z_pos: 0.085
lidar_yaw_angle: 180

# Node parameters
dt: 0.02
publish_odom: true

# Debug mode
debug_mode: false
plot_mode: false

# Standalone mode
standalone_mode: false
